### ISSUE TITLE
Incompatible with Ecto 3

### ISSUE THREAD

**Issue by mikehostetler** [ikeikeikeike/some-sitemap#28]
Hit an error when I upgraded a project with `sitemap` to Ecto 3:

```
==> sitemap
Compiling 19 files (.ex)

== Compilation error in file lib/sitemap/funcs.ex ==
** (CompileError)  deadlocked waiting on struct Ecto.DateTime


== Compilation error in file lib/sitemap/config.ex ==
** (CompileError)  deadlocked waiting on module Sitemap.Funcs


Compilation failed because of a deadlock between files.
The following files depended on the following modules:

   lib/sitemap/funcs.ex => Ecto.DateTime
  29   end
  lib/sitemap/config.ex => Sitemap.Funcs

Ensure there are no compile-time dependencies between those files and that the modules they reference exist and are correctly named

could not compile dependency :sitemap, "mix compile" failed. You can recompile this dependency with "mix deps.compile sitemap", update it with "mix deps.update sitemap" or clean it with "mix deps.clean sitemap"
```

**Comment by ssbb** [comment]
@ikeikeikeike just curious - why wontfix tag added? is there no plans to make it compatible with Ecto3?

**Comment by ssbb** [comment]
I think checking for Ecto major version should be enough here 🤔

**Comment by ssbb** [comment]
I just created PR #31 which fixes this issue.

**Comment by ikeikeikeike** [comment]
I think this issue was fixed by https://github.com/ikeikeikeike/sitemap/pull/32 and #31
